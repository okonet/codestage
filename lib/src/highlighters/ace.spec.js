/* eslint no-unused-expressions: 0 */
const { getThemes, getLanguages, loadTheme } = require('./ace')

describe('loadTheme', () => {
  it('should load CSS, strip CSS specificity and return a string', async () => {
    const { cssText } = await loadTheme('github')
    expect(cssText).toMatchSnapshot()
  })

  it('should not contain specificity', async () => {
    const { cssText } = await loadTheme('github')
    expect(cssText).not.toMatch('.ace-github .ace')
  })

  it('should match the shape', async () => {
    const theme = await loadTheme('github')
    expect(theme).toEqual(
      expect.objectContaining({
        cssText: expect.any(String),
        name: expect.any(String)
      })
    )
  })
})

describe('getThemes', () => {
  it('should return themes as a key values object', async () => {
    const themes = await getThemes()
    Object.keys(themes).forEach(key => expect(key).toEqual(themes[key].name))
  })

  it('should match the shape', async () => {
    const themes = await getThemes()
    Object.values(themes).forEach(theme => {
      expect(theme).toEqual(
        expect.objectContaining({
          caption: expect.any(String),
          cssText: expect.any(String),
          isDark: expect.any(Boolean),
          name: expect.any(String),
          theme: expect.any(String)
        })
      )
    })
  })
})

describe('getLanguages', () => {
  it('should return array of objects with names and paths to languages', async () => {
    expect(await getLanguages()).toMatchSnapshot()
  })
})
