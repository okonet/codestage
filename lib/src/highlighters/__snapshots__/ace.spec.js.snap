// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getLanguages should return array of objects with names and paths to languages 1`] = `
Array [
  "abap",
  "abc",
  "actionscript",
  "ada",
  "apache_conf",
  "applescript",
  "asciidoc",
  "assembly_x86",
  "autohotkey",
  "batchfile",
  "bro",
  "c9search",
  "c_cpp",
  "cirru",
  "clojure",
  "cobol",
  "coffee",
  "coldfusion",
  "csharp",
  "css",
  "curly",
  "d",
  "dart",
  "diff",
  "django",
  "dockerfile",
  "dot",
  "drools",
  "eiffel",
  "ejs",
  "elixir",
  "elm",
  "erlang",
  "forth",
  "fortran",
  "ftl",
  "gcode",
  "gherkin",
  "gitignore",
  "glsl",
  "gobstones",
  "golang",
  "groovy",
  "haml",
  "handlebars",
  "haskell",
  "haskell_cabal",
  "haxe",
  "hjson",
  "html",
  "html_elixir",
  "html_ruby",
  "ini",
  "io",
  "jack",
  "jade",
  "java",
  "javascript",
  "json",
  "jsoniq",
  "jsp",
  "jsx",
  "julia",
  "kotlin",
  "latex",
  "lean",
  "less",
  "liquid",
  "lisp",
  "live_script",
  "livescript",
  "logiql",
  "lsl",
  "lua",
  "luapage",
  "lucene",
  "makefile",
  "markdown",
  "mask",
  "matlab",
  "mavens_mate_log",
  "maze",
  "mel",
  "mips_assembler",
  "mipsassembler",
  "mushcode",
  "mysql",
  "nix",
  "nsis",
  "objectivec",
  "ocaml",
  "pascal",
  "perl",
  "pgsql",
  "php",
  "plain_text",
  "powershell",
  "praat",
  "prolog",
  "properties",
  "protobuf",
  "python",
  "r",
  "razor",
  "rdoc",
  "rhtml",
  "rst",
  "ruby",
  "rust",
  "sass",
  "scad",
  "scala",
  "scheme",
  "scss",
  "sh",
  "sjs",
  "smarty",
  "snippets",
  "soy_template",
  "space",
  "sql",
  "sqlserver",
  "stylus",
  "svg",
  "swift",
  "swig",
  "tcl",
  "tex",
  "text",
  "textile",
  "toml",
  "tsx",
  "twig",
  "typescript",
  "vala",
  "vbscript",
  "velocity",
  "verilog",
  "vhdl",
  "wollok",
  "xml",
  "xquery",
  "yaml",
]
`;

exports[`highlight should add gutter to the output 1`] = `
"<div class='ace_static_highlight' style='counter-reset:ace_line 
0'><div class='ace_line'><span class='ace_gutter ace_gutter-cell'>1</span><span class='ace_text ace_xml'>line</span>
</div><div class='ace_line'><span class='ace_gutter ace_gutter-cell'>2</span><span class='ace_text ace_xml'>line</span></div></div>"
`;

exports[`highlight should not add unnecessary new line 1`] = `
"<div class='ace_static_highlight' style='counter-reset:ace_line 
0'><div class='ace_line'><span class='ace_markup ace_list'>1. </span><span class='ace_list'>line</span>
</div><div class='ace_line'><span class='ace_markup ace_list'>2. </span><span class='ace_list'>line</span></div></div>"
`;

exports[`highlight should pad line numbers in the gutter depending on lines number 1`] = `
"<div class='ace_static_highlight' style='counter-reset:ace_line 
0'><div class='ace_line'><span class='ace_gutter ace_gutter-cell'> 1</span><span class='ace_text ace_xml'>line</span>
</div><div class='ace_line'><span class='ace_gutter ace_gutter-cell'> 2</span><span class='ace_text ace_xml'>line</span>
</div><div class='ace_line'><span class='ace_gutter ace_gutter-cell'> 3</span><span class='ace_text ace_xml'>line</span>
</div><div class='ace_line'><span class='ace_gutter ace_gutter-cell'> 4</span><span class='ace_text ace_xml'>line</span>
</div><div class='ace_line'><span class='ace_gutter ace_gutter-cell'> 5</span><span class='ace_text ace_xml'>line</span>
</div><div class='ace_line'><span class='ace_gutter ace_gutter-cell'> 6</span><span class='ace_text ace_xml'>line</span>
</div><div class='ace_line'><span class='ace_gutter ace_gutter-cell'> 7</span><span class='ace_text ace_xml'>line</span>
</div><div class='ace_line'><span class='ace_gutter ace_gutter-cell'> 8</span><span class='ace_text ace_xml'>line</span>
</div><div class='ace_line'><span class='ace_gutter ace_gutter-cell'> 9</span><span class='ace_text ace_xml'>line</span>
</div><div class='ace_line'><span class='ace_gutter ace_gutter-cell'>10</span><span class='ace_text ace_xml'>line</span>
</div><div class='ace_line'><span class='ace_gutter ace_gutter-cell'>11</span></div></div>"
`;

exports[`highlight should return HTML with proper class names 1`] = `
"<div class='ace_static_highlight' style='counter-reset:ace_line 
0'><div class='ace_line'><span class='ace_keyword'>async</span> <span class='ace_storage ace_type'>function</span> <span class='ace_entity ace_name ace_function'>html2rtf</span><span class='ace_paren ace_lparen'>(</span><span class='ace_paren ace_rparen'>)</span> <span class='ace_paren ace_lparen'>{</span><span class='ace_paren ace_rparen'>}</span></div></div>"
`;

exports[`loadTheme should load CSS, strip CSS specificity and return a string 1`] = `".ace_gutter {background: #e8e8e8;color: #AAA;}.ace_static_highlight  {background: #fff;color: #000;}.ace_keyword {font-weight: bold;}.ace_string {color: #D14;}.ace_variable.ace_class {color: teal;}.ace_constant.ace_numeric {color: #099;}.ace_constant.ace_buildin {color: #0086B3;}.ace_support.ace_function {color: #0086B3;}.ace_comment {color: #998;font-style: italic;}.ace_variable.ace_language  {color: #0086B3;}.ace_paren {font-weight: bold;}.ace_boolean {font-weight: bold;}.ace_string.ace_regexp {color: #009926;font-weight: normal;}.ace_variable.ace_instance {color: teal;}.ace_constant.ace_language {font-weight: bold;}.ace_cursor {color: black;}.ace_static_highlight.ace_focus .ace_marker-layer .ace_active-line {background: rgb(255, 255, 204);}.ace_marker-layer .ace_active-line {background: rgb(245, 245, 245);}.ace_marker-layer .ace_selection {background: rgb(181, 213, 255);}.ace_static_highlight.ace_multiselect .ace_selection.ace_start {box-shadow: 0 0 3px 0px white;}.ace_static_highlight.ace_nobold .ace_line > span {font-weight: normal !important;}.ace_marker-layer .ace_step {background: rgb(252, 255, 0);}.ace_marker-layer .ace_stack {background: rgb(164, 229, 101);}.ace_marker-layer .ace_bracket {margin: -1px 0 0 -1px;border: 1px solid rgb(192, 192, 192);}.ace_gutter-active-line {background-color : rgba(0, 0, 0, 0.07);}.ace_marker-layer .ace_selected-word {background: rgb(250, 250, 255);border: 1px solid rgb(200, 200, 250);}.ace_invisible {color: #BFBFBF}.ace_print-margin {width: 1px;background: #e8e8e8;}.ace_indent-guide {background: url(\\"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAACCAYAAACZgbYnAAAAE0lEQVQImWP4////f4bLly//BwAmVgd1/w11/gAAAABJRU5ErkJggg==\\") right repeat-y;}"`;
